### ISSUE TITLE
Build failure on Sierra (10.12.2) due to libqwt-dev dependency

### ISSUE THREAD

**Issue by starthal** [osrf/homebrew-simulation#215]
```
brew install gazebo8
==> Installing gazebo8 from osrf/simulation
==> Downloading http://gazebosim.org/distributions/gazebo/releases/gazebo-8.0.0.tar.bz2
Already downloaded: /Users/salbert/Library/Caches/Homebrew/gazebo8-8.0.0.tar.bz2
==> Downloading https://gist.githubusercontent.com/scpeters/9199370/raw/afe595587e38737c537124a3652db99de026c272/brew_python_fix.patch
Already downloaded: /Users/salbert/Library/Caches/Homebrew/gazebo8--patch-c4774f64c490fa03236564312bd24a8630963762e25d98d072e747f0412df18e.patch
==> Patching
==> Applying brew_python_fix.patch
patching file deps/opende/CMakeLists.txt
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/opt/brew/Cellar/gazebo8/8.0.0_2 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev
Last 15 lines from /Users/salbert/Library/Logs/Homebrew/gazebo8/01.cmake:
-- 	Graphviz not found, Model editor's schematic view will be disabled.
-- 	OpenSource Virtual Reality (OSVR) support will be disabled.
-- END BUILD WARNINGS

-- BUILD ERRORS: These must be resolved before compiling.
-- 	Missing: libqwt-dev. Required for plotting.
-- END BUILD ERRORS

CMake Error at CMakeLists.txt:303 (message):
  Errors encountered in build.  Please see the BUILD ERRORS above.


-- Configuring incomplete, errors occurred!
See also "/tmp/gazebo8-20170203-95067-1g1mcoo/gazebo-8.0.0/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/gazebo8-20170203-95067-1g1mcoo/gazebo-8.0.0/build/CMakeFiles/CMakeError.log".
```

qwt-6.1.3_3 is installed according to Homebrew.

I would post the CMake logs, but I don't know how to keep Homebrew from deleting them.

**Comment by scpeters** [comment]
can you try:

~~~
brew install gazebo8 --verbose
brew gist-logs gazebo8
~~~

that should upload the logs automatically. thanks!

**Comment by scpeters** [comment]
Also, can you include the output of `brew info qwt`?

**Comment by starthal** [comment]
[Logs](https://gist.github.com/anonymous/5a1ec901c4899e5797083dc01b737c76)

`brew info qwt`:

```
qwt: stable 6.1.3
Qt Widgets for Technical Applications
http://qwt.sourceforge.net/
/opt/brew/Cellar/qwt/6.1.3_3 (1,669 files, 18.7M) *
  Built from source on 2017-02-02 at 16:40:43
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/qwt.rb
==> Dependencies
Required: qt5 ✔
==> Options
--with-qwtmathml
	Build the qwtmathml library
--without-plugin
	Skip building the Qt Designer plugin
```

**Comment by starthal** [comment]
I did try moving MacPorts out of the way as well -- no effect.

**Comment by scpeters** [comment]
It looks like gazebo doesn't handle an alternative install prefix of `qwt`:

* https://bitbucket.org/osrf/gazebo/src/56be256c4b/cmake/SearchForStuff.cmake#SearchForStuff.cmake-753

It searches in either `/usr` or `/usr/local`

**Comment by scpeters** [comment]
Can you try applying this patch to the gazebo8 formula:

~~~
diff --git a/gazebo8.rb b/gazebo8.rb
index d17eb24..97a859b 100644
--- a/gazebo8.rb
+++ b/gazebo8.rb
@@ -63,6 +63,8 @@ class Gazebo8 < Formula
     ENV.m64
 
     cmake_args = std_cmake_args
+    cmake_args << "-DQWT_WIN_INCLUDE_DIR=${HOMEBREW_PREFIX}/lib/qwt.framework/Headers"
+    cmake_args << "-DQWT_WIN_LIBRARY_DIR=${HOMEBREW_PREFIX}/lib/qwt.framework"
 
     mkdir "build" do
       system "cmake", "..", *cmake_args
~~~

**Comment by scpeters** [comment]
I'll make a pull request with this if it works for you.

**Comment by starthal** [comment]
It didn't work with `${HOMEBREW_PREFIX}` --  that must not be expanding properly. But when I replaced that with my hardcoded prefix it worked.

**Comment by starthal** [comment]
I think it needs to be expanded within Ruby with `#{HOMEBREW_PREFIX}` instead.

**Comment by starthal** [comment]
I can confirm that it builds and runs with that change. I do need to run it with `DYLD_FALLBACK_LIBRARY_PATH=$(brew --prefix)/lib` but I suspect that's a different issue.

**Comment by scpeters** [comment]
ok, thanks for testing. I'll make a PR using the `#{}` syntax

**Comment by scpeters** [comment]
I merged that fix in #216. Please re-open if that fix is not sufficient (excluding the `DYLD_FALLBACK_LIBRARY_PATH` issue)

**Comment by starthal** [comment]
Thank you!
